icx/src/main.rs

```diff
@@ -219,14 +233,80 @@ fn create_identity(maybe_pem: Option<PathBuf>) -> impl Identity {
     }
 }
 
+#[derive(Error, Debug)]
+enum SerializeError {
+    #[error("")]
+    Success,
+}
+
+struct SerializingFacade;
+
+impl agent::ReplicaV1Facade for SerializingFacade {
+    fn read<'a>(
+        &'a self,
+        envelope: Vec<u8>,
+    ) -> Pin<Box<dyn Future<Output = Result<Vec<u8>, AgentError>> + Send + 'a>> {
+        async fn run(_: &SerializingFacade, envelope: Vec<u8>) -> Result<Vec<u8>, AgentError> {
+            print!("read\n\n{}", hex::encode(envelope));
```

Comment:
Are these `print` statements intentional and meant to not be `eprint`?

Reply:
(I see the separated println("result") / eprintln(result) below)

Reply:
Yes, tthis facade is meant to serialize data to STDOUT, so it could be piped to a file or another process.